src/main/java/net/minecraftforge/java_provisioner/JavaVersion.java

```diff
@@ -62,15 +65,15 @@ private static int toInt(String s) {
         return val;
     }
 
-    public static JavaVersion nullableParse(String s) {
+    public @Nullable static JavaVersion nullableParse(String s) {
         try {
             return parse(s);
         } catch (IllegalArgumentException e) {
             return null;
         }
     }
 
-    public static JavaVersion parse(String s) {
+    public static JavaVersion parse(@UnknownNullability String s) {
```

Comment:
Mismatched null annotations: this has one on the string parameter but nullableParse(String s) doesn't.

This method always throws when s is null, so should it be annotated as unknown nullability or should it be left unannotated to indicate it throws when given nulls? Or does JetBrains null annotations apply only to NPEs, and not throwing other kinds of exceptions when encountering nulls?